pyo3-build-config/src/impl_.rs

```diff
@@ -456,6 +456,28 @@ print("mingw", get_platform().startswith("mingw"))
         })
     }
 
+    #[allow(clippy::unnecessary_wraps)]
+    pub fn fixup_import_libs(&mut self) -> Result<()> {
+        // Auto generate python3.dll import libraries for Windows targets.
+        #[cfg(feature = "python3-dll-a")]
+        {
+            let target = target_triple_from_env();
+            if self.lib_name.is_none() && target.operating_system == OperatingSystem::Windows {
+                self.lib_name = Some(default_lib_name_windows(
+                    self.version,
+                    self.implementation,
+                    self.abi3,
+                    false,
+                ));
+            }
```

Comment:
I think this can be done unconditionally, because `lib_name` has to be set to _something_ on Windows, no matter whether `python3-dll-a` is enabled or not.